### PR TITLE
Deploy prow-monitoring stack with post-test-infra-deploy-prow

### PR DESCRIPTION
**Pull request by hongkailiu**
https://github.com/kubernetes/test-infra/issues/12854

### CODE REVIEW

File: config/jobs/kubernetes/test-infra/test-infra-trusted.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ postsubmits:
   kubernetes/test-infra:
   - name: post-test-infra-deploy-prow
     cluster: test-infra-trusted
-    run_if_changed: 'prow/cluster/'
+    run_if_changed: '^prow/(cluster|monitoring)/'
```
Comment (fejta) [inline config/jobs/kubernetes/test-infra/test-infra-trusted.yaml]: Instead can we please move all these files into prow/cluster/monitoring?

File: prow/deploy.sh [hunk 2]
```diff
@@ -88,4 +88,6 @@ for s in {5..1}; do
     sleep 1s
 done
 bazel run //prow/cluster:production.apply --platforms=@io_bazel_rules_go//go/toolchain:linux_amd64
```
Comment (fejta) [inline prow/deploy.sh]: Can we please make this k8s_objects() target include //prow/monitoring:produciton.apply and //prow/monitoring/mixings/dashboards_out:grafana-configmaps.apply?

We should be able to update prow with a single call, not 3.

File: config/jobs/kubernetes/test-infra/test-infra-trusted.yaml (hunk 1 above)
Comment (hongkailiu) [reply config/jobs/kubernetes/test-infra/test-infra-trusted.yaml]: Done

File: prow/deploy.sh (hunk 2 above)
Comment (hongkailiu) [reply prow/deploy.sh]: I need more hints to merge the targets. If I understand it correctly, `//prow/cluster:production` accepts only `components`. However,

* some of objects use `k8s_configmap` function to generate the `k8s_object`.
* `components` only works with yaml files on the current folder.

Instead, use `multirun` to make a bit nicer.

https://github.com/kubernetes/test-infra/blob/c8298ea38c506a8c9d1688ffd0370b95551dd969/prow/cluster/BUILD.bazel#L47-L55

**Pull request merged**